applications/HDF5Application/custom_io/hdf5_container_component_io.cpp

```diff
@@ -389,477 +96,331 @@ ContainerComponentIO<TContainerType, TContainerItemType, TComponents...>::Contai
-    const std::string& rComponentName, Targs&... args)
+template <class TContainerType, class TContainerDataIO, class... TComponents>
+void ContainerComponentIO<TContainerType, TContainerDataIO, TComponents...>::Write(
+    const TContainerType& rLocalContainer,
+    const TContainerDataIO& rContainerDataIO,
+    const Parameters Attributes)
 {
-    RegisteredComponentLookup<Flags>(rComponentName).Execute<WriteConditionComponent>(args...);
-}
+    KRATOS_TRY;
 
-// Adding condition other variable WriteRegisteredComponent template definition
-template <>
-template <typename... Targs>
-void ContainerComponentIO<ConditionsContainerType,
-                          ConditionType,
-                          Variable<array_1d<double, 3>>,
-                          Variable<double>,
-                          Variable<int>,
-                          Variable<Vector<double>>,
-                          Variable<Matrix<double>>>::WriteRegisteredComponent(const std::string& rComponentName,
-                                                                              Targs&... args)
-{
-    RegisteredComponentLookup<Variable<array_1d<double, 3>>, Variable<double>, Variable<int>,
-                              Variable<Vector<double>>, Variable<Matrix<double>>>(rComponentName)
-        .Execute<WriteConditionComponent>(args...);
-}
+    if (mComponentNames.size() == 0) {
+        return;
+    }
 
-// Adding condition flags ReadRegisteredComponent template definition
-template <>
-template <typename... Targs>
-void ContainerComponentIO<ConditionsContainerType, ConditionType, Flags>::ReadRegisteredComponent(
-    const std::string& rComponentName, Targs&... args)
-{
-    RegisteredComponentLookup<Flags>(rComponentName).Execute<ReadConditionComponent>(args...);
-}
+    WriteInfo info;
 
-// Adding condition other variable ReadRegisteredComponent template definition
-template <>
-template <typename... Targs>
-void ContainerComponentIO<ConditionsContainerType,
-                          ConditionType,
-                          Variable<array_1d<double, 3>>,
-                          Variable<double>,
-                          Variable<int>,
-                          Variable<Vector<double>>,
-                          Variable<Matrix<double>>>::ReadRegisteredComponent(const std::string& rComponentName,
-                                                                             Targs&... args)
-{
-    RegisteredComponentLookup<Variable<array_1d<double, 3>>, Variable<double>, Variable<int>,
-                              Variable<Vector<double>>, Variable<Matrix<double>>>(rComponentName)
-        .Execute<ReadConditionComponent>(args...);
-}
+    // Write each variable.
+    for (const auto& r_component_name : mComponentNames) {
+        const bool is_component_written = (... || (WriteComponentData<TComponents>(
+                        r_component_name, rContainerDataIO, rLocalContainer, Attributes.Clone(), info)));
```

Comment:
not sure about this one, but shouldn't it be `&&`?


```suggestion
        const bool is_component_written = (... && (WriteComponentData<TComponents>(
                        r_component_name, rContainerDataIO, rLocalContainer, Attributes.Clone(), info)));
```

Reply:
nevermind I think I understand now ...

this loops over the types and checks whether the name is registered in the corresponding instantiation of `KratosComponents`. Most return `false` and only one should return `true` (for the type of the requested variable). Is that correct?

Reply:
You are correct 👍